include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -159,7 +169,8 @@ class PreconditionerBoomerAMG : public PreconditionerBase<Number>
   {
 #ifdef DEAL_II_WITH_PETSC
     // clear content of matrix since the next calculate_system_matrix-commands add their result
-    system_matrix = 0.0;
+    if(system_matrix.m() > 0)
```

Comment:
The meaning of this if is not a 100% clear to me. Did you introduce it because of the sub-communicator or is it a general optimization? Maybe a comment would help the non-expert here.

Reply:
I added the comment in my fc0d323d commit (sorry I was most effective to work on all comments at the same time and could not easily split it up afterwards).